setup.py

```diff
@@ -27,6 +27,7 @@
         'jsonschema',
         'python-json-logger',
         'traitlets',
-        'ruamel.yaml'
+        'ruamel.yaml',
+        'sphinx-jsonschema'
```

Comment:
I don't think this should be a required dependency, since we do not need it at runtime. And sphinx is large!

Maybe we can make the sphinx package its own package, living in a subdirectory here?